tuf/client/client.go

```diff
@@ -439,18 +446,19 @@ func (c *Client) downloadTargets(role string) error {
 	return nil
 }
 
-func (c *Client) downloadSigned(role string, size int64, expectedSha256 []byte) ([]byte, *data.Signed, error) {
-	rolePath := utils.ConsistentName(role, expectedSha256)
+func (c *Client) downloadSigned(role string, size int64, expectedHashes data.Hashes) ([]byte, *data.Signed, error) {
+	rolePath := utils.ConsistentName(role, expectedHashes["sha256"])
 	raw, err := c.remote.GetMeta(rolePath, size)
 	if err != nil {
 		return nil, nil, err
 	}
-	if expectedSha256 != nil {
-		genHash := sha256.Sum256(raw)
-		if !bytes.Equal(genHash[:], expectedSha256) {
+
+	if len(expectedHashes) != 0 {
```

Comment:
The problem is that `len(expectedHashes)` on a nil map is a valid operation and also returns length 0. We want a clear distinction between "I am explicitly saying there are no hashes to check" and "the snapshot/timestamp contained an empty hashes entry (probably as the result of malicious behaviour)". There are other checks that need to be done elsewhere because the hashes entry could itself be `null` in the JSON, but this particular method is generic and should distinguish the 2 above cases.